packages/twenty-server/src/modules/timeline/services/timeline-activity.service.ts

```diff
@@ -61,56 +71,71 @@ export class TimelineActivityService {
         event.objectMetadata.nameSingular,
       )
     ) {
-      const linkedObjects = await this.handleLinkedObjects(event);
-
-      return linkedObjects;
+      return await this.handleLinkedObjects(event, workspaceId, eventName);
     }
 
     return [event];
   }
 
   private async handleLinkedObjects(
-    event: ObjectRecordBaseEventWithNameAndWorkspaceId,
+    event: ObjectRecordBaseEvent,
+    workspaceId: string,
+    eventName: string,
   ) {
-    const dataSourceSchema = this.workspaceDataSourceService.getSchemaName(
-      event.workspaceId,
-    );
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
 
     switch (event.objectMetadata.nameSingular) {
       case 'noteTarget':
-        return this.processActivityTarget(event, dataSourceSchema, 'note');
+        return this.processActivityTarget(
+          event,
+          dataSourceSchema,
+          'note',
+          eventName,
+          workspaceId,
+        );
       case 'taskTarget':
-        return this.processActivityTarget(event, dataSourceSchema, 'task');
+        return this.processActivityTarget(
```

Comment:
see here it becomes a nightmare to have 5 arguments. Also help making things optional without having to re-order